components/org.wso2.carbon.micro.integrator.extensions/org.wso2.carbon.micro.integrator.management.apis/src/main/java/org/wso2/carbon/micro/integrator/management/apis/TemplateResource.java

```diff
@@ -0,0 +1,200 @@
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.micro.integrator.management.apis;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.apache.synapse.MessageContext;
+import org.apache.synapse.config.SynapseConfiguration;
+import org.apache.synapse.core.axis2.Axis2MessageContext;
+import org.apache.synapse.endpoints.Template;
+import org.apache.synapse.mediators.template.TemplateMediator;
+import org.json.JSONArray;
+import org.json.JSONObject;
+import org.wso2.carbon.inbound.endpoint.internal.http.api.APIResource;
+
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+
+/**
+ * Represents template resources defined in the synapse configuration.
+ **/
+public class TemplateResource extends APIResource {
+
+    private static final Log log = LogFactory.getLog(TemplateResource.class);
+
+    /* Name of the JSON list of endpoint templates */
+    private static final String ENDPOINT_TEMPLATE_LIST = "endpointTemplateList";
+    /* Name of the JSON list of sequence templates */
+    private static final String SEQUENCE_TEMPLATE_LIST = "sequenceTemplateList";
+    /* Template type parameter */
+    private static final String TEMPLATE_TYPE_PARAM = "templateType";
+    /* Template name parameter */
+    private static final String TEMPLATE_NAME_PARAM = "templateName";
+    /* Possible values for TEMPLATE_TYPE_PARAM  */
+    private static final String ENDPOINT_TEMPLATE_TYPE = "endpoint";
+    private static final String SEQUENCE_TEMPLATE_TYPE = "sequence";
+
+    public TemplateResource(String urlTemplate) {
+        super(urlTemplate);
+    }
+
+    @Override
+    public Set<String> getMethods() {
+        Set<String> methods = new HashSet<>();
+        methods.add("GET");
+        methods.add("POST");
+        return methods;
+    }
+
+    @Override
+    public boolean invoke(MessageContext messageContext) {
+        org.apache.axis2.context.MessageContext axis2MessageContext =
+                ((Axis2MessageContext) messageContext).getAxis2MessageContext();
+        SynapseConfiguration synapseConfiguration = messageContext.getConfiguration();
+        if (Objects.isNull(axis2MessageContext) || Objects.isNull(synapseConfiguration)) {
+            return false;
+        }
+
+        String templateTypeParam = Utils.getQueryParameter(messageContext, TEMPLATE_TYPE_PARAM);
+        if (Objects.nonNull(templateTypeParam)) {
+            String templateNameParam = Utils.getQueryParameter(messageContext, TEMPLATE_NAME_PARAM);
```

Comment:
Do we need the template type also even if we pass the template name in query param to retrieve a particular template.

Reply:
Retrieving endpoint templates and sequence templates are done separately. Therefore template type is mandatory.

Reply:
Isn't that an internal implementation, but for the user they should be able to retrieve just by giving the name since it is supposed to be unique. Shall we handle that internally and expect only the template name in query param ?

Reply:
In the existing UI (EI), the templates are listed by type. So do we really have to allow to search just by name in this context?

Reply:
I think it is because we have two underlying objects Template & TemplateMediator in configuration. Anyway if the user just passes the name, it is a sufficient key to retrieve the template without type. We can just do both retrievals in code level https://github.com/wso2/micro-integrator/pull/248/files#diff-f53d90bd86f38cd7aa76b441faf51374R168 && https://github.com/wso2/micro-integrator/pull/248/files#diff-f53d90bd86f38cd7aa76b441faf51374R171 and return the necessary template by doing some filtering.  Since it s just a simple code level fix and would improve UX , I thought it would be better . WDYT ?